flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_Boarding.js

```diff
@@ -0,0 +1,157 @@
+function airplaneCanBoard() {
+    const busDC2 = SimVar.GetSimVarValue("L:A32NX_ELEC_DC_2_BUS_IS_POWERED", "Bool");
+    const busDCHot1 = SimVar.GetSimVarValue("L:A32NX_ELEC_DC_HOT_1_BUS_IS_POWERED", "Bool");
+    const gs = SimVar.GetSimVarValue("GPS GROUND SPEED", "knots");
+    const isOnGround = SimVar.GetSimVarValue("SIM ON GROUND", "Bool");
+    const eng1Running = SimVar.GetSimVarValue("ENG COMBUSTION:1", "Bool");
+    const eng2Running = SimVar.GetSimVarValue("ENG COMBUSTION:2", "Bool");
+
+    return !(gs > 0.1 || eng1Running || eng2Running || !isOnGround || (!busDC2 && !busDCHot1));
+}
+
+class A32NX_Boarding {
+    constructor() {
+        this.boardingState = "finished";
+        this.time = 0;
+        const payloadConstruct = new A32NX_PayloadConstructor();
+        this.paxStations = payloadConstruct.paxStations;
+        this.payloadStations = payloadConstruct.payloadStations;
+    }
+
+    async init() {
+        // Set default pax (0)
+        await this.setPax(0);
+        await this.loadPayload();
+        await this.loadCargoZero();
+    }
+
+    async fillStation(station, paxToFill) {
+        const pax = Math.min(paxToFill, station.seats);
+        station.pax = pax;
+
+        await SimVar.SetSimVarValue(`L:${station.simVar}`, "Number", parseInt(pax));
+    }
+
+    async setPax(numberOfPax) {
+        let paxRemaining = parseInt(numberOfPax);
+
+        async function fillStation(station, paxToFill) {
+            const pax = Math.min(paxToFill, station.seats);
+            station.pax = pax;
+
+            await SimVar.SetSimVarValue(`L:${station.simVar}`, "Number", parseInt(pax));
+
+            paxRemaining -= pax;
+        }
+
+        await fillStation(this.paxStations['rows22_29'], paxRemaining);
+        await fillStation(this.paxStations['rows14_21'], paxRemaining);
+
+        const remainingByTwo = Math.trunc(paxRemaining / 2);
+        await fillStation(this.paxStations['rows1_6'], remainingByTwo);
+        await fillStation(this.paxStations['rows7_13'], paxRemaining);
+
+        return;
+    }
+
+    async loadPayload() {
+        const MAX_SEAT_AVAILABLE = 174;
+        const PAX_WEIGHT = 84;
+        const BAG_WEIGHT = 20;
+
+        const currentPaxWeight = PAX_WEIGHT + BAG_WEIGHT;
```

Comment:
Can you clarify if the bag weight is being added to the passenger load station weight? If so, it will result in the max weight for that station being reached before the max number of passengers is reached. The max load for each station in the flight model.cfg file is based on the passenger weight only. Unless that is an oversight, shouldn't the 20 kg of bags be put in the AFT Bulk/Loose baggage station?

Reply:
I had a look with Simvar Watcher into the inner workings of constructing and filling these payload stations. There is definitely a couple of new errors that I've spotted:

- Pilot/FO weights are not being accounted
- MCDU init causes payload values (not fuel) to be 0 when spawning. That is, initial payload from MSFS Fuel UI in the aircraft selection panel is always reset. (this is more of a beneficial bug, since the weights ultimately interfere with the actual load from W/B)
- The stations for economy pax are from 2-5, in reality the filling happens in stations 3-6. This doesn't have an effect on the overall weight, but does affect the distribution.
- You were also correct about the 20KG baggage being loaded into the PAX station, which is ultimately loaded above capacity. I am considering removing this 20KG per pax and putting it in a new function to load the AFT Baggage and FWD Baggage stations in the particular order.
- Another problem I noticed is, even with the limits specified in flightmodel.cfg, we are able to overload the stations way beyond their capacities. I'm thinking of making this more robust from our end instead of relying on the Fuel UI/in-sim system.

All these issues are not affecting the current implementation though it can and will be corrected, which is missing a few weights from the cockpit and the cargo stations, and should be fixed in a day or two. Thanks for going through these once again!